src/main/java/step4/Cars.java

```diff
@@ -0,0 +1,28 @@
+package step4;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Cars implements Iterable<Car> {
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public Cars winners() {
+        Map<Integer, List<Car>> result = this.cars.stream().collect(Collectors.groupingBy(Car::location));
```

Comment:
stream 활용 👍👍

추가로 가독성을 위해 온점(.)을 한 라인에 한 개만 찍는 것은 어떨까요?
```suggestion
        Map<Integer, List<Car>> result = this.cars.stream()
            .collect(Collectors.gruopingBy(Car::locatoin);        
```